src/filter/glitch0r/glitch0r.c

```diff
@@ -0,0 +1,347 @@
+void fastsrand(uint32_t seed) { randval = seed; };
+
+struct glitch0r_state // helps to save time when allocating in a loop
+{
+    unsigned int currentBlock;
+    unsigned int currentPos;
+    unsigned int currentY;
+    unsigned int blkShift;
+
+    uint32_t distortionSeed1;
+    uint32_t distortionSeed2;
+
+    short int howToDistort1;
+    short int howToDistort2;
+    short int passThisLine;
+} g0r_state;
+
+typedef struct glitch0r_instance
+{
+    unsigned int width;
+    unsigned int height;
+
+    unsigned int maxBlockSize;
+    unsigned int maxBlockShift;
+
+    short int colorGlitchIntensity;
+    short int doColorDistortion;
+    short int glitchChance;
+} glitch0r_instance_t;
+
+
+inline unsigned int rnd (unsigned int min, unsigned int max)
+{
+    return fastrand() % (max - min + 1) + min;
+}
+
+inline void glitch0r_state_reset(glitch0r_instance_t *inst)
+{
+    g0r_state.currentPos = 0;
+    g0r_state.currentBlock = rnd(1, inst->maxBlockSize);
+    g0r_state.blkShift = rnd(1, inst->maxBlockShift);
+    g0r_state.passThisLine = (inst->glitchChance < rnd(1, 101)) ? 1 : 0;
+
+    if (inst->doColorDistortion)
+    {
+        g0r_state.distortionSeed1 = rnd(0x00000000, 0x00ffffff);
+        g0r_state.distortionSeed2 = rnd(0x00000000, 0x00ffffff);
+        g0r_state.howToDistort1 = rnd (0, inst->colorGlitchIntensity);
+        g0r_state.howToDistort2 = rnd (0, inst->colorGlitchIntensity);
+    }
+}
+
+inline void glitch0r_pixel_dist0rt (uint32_t *pixel,
+            uint32_t distortionSeed, short int howToDistort)
+{
+    // Choose from five levels of madness:
+    switch (howToDistort)
+    {
+        case 0 : break; // ok, let this pixel live (just shift)
+
+        case 1 : // lightest distortion: just invert
+            *(pixel) = ~*(pixel);
+            break;
+
+        case 2 : // add some unneeded colors
+            *(pixel) = distortionSeed | *(pixel);
+            break;
+
+        case 3 : // change some colors
+            *(pixel) = distortionSeed ^ *(pixel);
+             break;
+
+        case 4 : // oh shi...
+            *(pixel) = distortionSeed & *(pixel);
+            break;
+    }  
+}
+
+int f0r_init()
+{
+    fastsrand((uint32_t)time(0));
+    return 1;
+}
+
+void f0r_deinit()
+{ /* no initialization required */ }
+
+void f0r_get_plugin_info(f0r_plugin_info_t* glitch0rInfo)
+{
+    glitch0rInfo->name = "Glitch0r";
+    glitch0rInfo->author = "IDENT Software";
+    glitch0rInfo->plugin_type = F0R_PLUGIN_TYPE_FILTER;
+    glitch0rInfo->color_model = F0R_COLOR_MODEL_RGBA8888;
+    glitch0rInfo->frei0r_version = FREI0R_MAJOR_VERSION;
+    glitch0rInfo->major_version = 0; 
+    glitch0rInfo->minor_version = 1; 
+    glitch0rInfo->num_params =  4; 
+    glitch0rInfo->explanation = "Adds glitches and block shifting";
+}
+
+void f0r_get_param_info(f0r_param_info_t* info, int param_index)
+{
+    switch(param_index) {
+        case 0:
+        {
+            info->name = "GlitchFrequency";
```

Comment:
This is not a big deal, but some frei0r plugin hosts expose these names as labels in the UI. Thus, spaces are preferred over PascalCase.